From 1efd29280029a997ff96631fdbafd75094557615 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Tue, 24 Nov 2020 16:40:06 +0100
Subject: [PATCH 1/1] add parenthesis around expression

Add parenthesis around expression before casting to boolean. This fixes:

main-window.vala:202.8-202.57: error: Condition must be boolean
			if ((bool) event.changed_mask & Gdk.WindowState.STICKY && this.visible) {
			    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
main-window.vala:204.9-204.73: error: Equality operation: `bool' and `Gdk.WindowState' are incompatible
				if (((bool) event.new_window_state & Gdk.WindowState.STICKY) == false)
				    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 src/main-window.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main-window.vala b/src/main-window.vala
index 8155ec1..a0755a9 100644
--- a/src/main-window.vala
+++ b/src/main-window.vala
@@ -199,9 +199,9 @@ namespace Xfmpc {
    	   	   	 * window-state-event signal, so here we take the value only if
    	   	   	 * the window is visible
    	   	   	 **/
-			if ((bool) event.changed_mask & Gdk.WindowState.STICKY && this.visible) {
+			if ((bool) (event.changed_mask & Gdk.WindowState.STICKY) && this.visible) {
 				bool sticky;
-				if (((bool) event.new_window_state & Gdk.WindowState.STICKY) == false)
+				if (((bool) (event.new_window_state & Gdk.WindowState.STICKY)) == false)
 					sticky = false;
 				else
 					sticky = true;
